funnel/views/notification.py

```diff
@@ -546,8 +546,34 @@ def dispatch_transport_sms(user_notification, view):
     )
 
 
+@rqjob
+@transport_worker_wrapper
+def dispatch_transport_whatsapp(user_notification, view):
+    if not user_notification.user.main_notification_preferences.by_transport(
+        'whatsapp'
+    ):
+        # Cancel delivery if user's main switch is off. This was already checked, but
+        # the worker may be delayed and the user may have changed their preference.
+        user_notification.messageid_whatsapp = 'cancelled'
+        return
+    user_notification.messageid_whatsapp = sms.send(
+        str(view.transport_for('sms')), view.sms_with_unsubscribe()
```

Comment:
`view.transport_for('whatsapp')` and `view.whatsapp()`.

The `whatsapp` method has to be added to the base RenderNotification class if not already present. The default implementation can simply use `return self.text()`